_posts/2020-01-06-using-strimzi-with-amazon-nlb-loadbalancers.md

```diff
@@ -0,0 +1,348 @@
+* Create NLB and configure it to route the data to the node port services
+* Configure the advertised addresses in Apache Kafka to use the NLB address instead of the node addresses and node ports
+
+Let's look in detail on each of these steps.
+
+### Exposing Apache Kafka with node ports
+
+First, we will deploy a regular Kafka cluster with node port access:
+
+```yaml
+    # ...
+    listeners:
+      # ...
+      external:
+        type: nodeport
+        tls: false
+    # ...
+```
+
+We will have to note the node port numbers of the node port services, because we will need them in the next step:
+
+```
+kubectl get service
+NAME                                  TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)                      AGE
+my-cluster-kafka-0                    NodePort    10.96.84.161   <none>        9094:31208/TCP               35m
+my-cluster-kafka-1                    NodePort    10.96.63.131   <none>        9094:31983/TCP               35m
+my-cluster-kafka-2                    NodePort    10.96.35.251   <none>        9094:32279/TCP               35m
+my-cluster-kafka-bootstrap            ClusterIP   10.96.17.28    <none>        9091/TCP,9092/TCP,9093/TCP   35m
+my-cluster-kafka-brokers              ClusterIP   None           <none>        9091/TCP,9092/TCP,9093/TCP   35m
+my-cluster-kafka-external-bootstrap   NodePort    10.96.187.42   <none>        9094:32200/TCP               35m
+my-cluster-zookeeper-client           ClusterIP   10.96.58.200   <none>        2181/TCP                     36m
+my-cluster-zookeeper-nodes            ClusterIP   None           <none>        2181/TCP,2888/TCP,3888/TCP   36m
+```
+
+In my case, the node ports are:
+* `32200`
+* `31208`
+* `31983`
+* `32279`
+
+### Creating and configuring the NLB
+
+Next, we have to create the Network load balancer.
+In the Amazon AWS console, go to the _Load Balancer_ section of the EC2 dashboard and create a new load balancer with the _Create Load Balancer_ button.
+In the wizard, select a _Network Load Balancer_:
+
+[![Selecting the load balancer type]({{ "/assets/2020-01-06-nlb-create-wizard-type.png" }})](/assets/2020-01-06-nlb-create-wizard-type.png)
+
+And configure the details of the load balancer:
+* Pick your own name for the load balancer. 
+I named it `my-cluster-kafka`.
+* Create one listener for each broker plus one additional as bootstrap. 
+All listeners should use the TCP protocol. 
+My Kafka cluster has 3 brokers, so I configured 4 listeners on ports `9092`, `19092`, `29092` and `39092`.
+* Select the right VPC where your Kubernetes cluster runs and select all availability zones which it is using.
+
+Once you are finished, click _Next: ..._.
+
+[![Configure listeners, VPC and subnets]({{ "/assets/2020-01-06-nlb-create-wizard-listeners.png" }})](/assets/2020-01-06-nlb-create-wizard-listeners.png)
+
+We can skip the security configuration since we do not use any TLS listeners and just click _Next: ..._ again to move to routing configuration:
```

Comment:
```suggestion
We can skip the security configuration since we do not use any TLS listeners and just click _Next: ..._ again to move to the routing configuration:
```